player/command.c

```diff
@@ -6061,7 +6061,7 @@ static void cmd_loadfile(void *p)
 
     void *tmp = talloc_new(NULL);
 
-    if (!files || !files[0] || !files[0][0]) {
+    if ((!files || !files[0] || !files[0][0]) && mp_input_is_interactive_cmd(cmd->cmd)) {
```

Comment:
I do like the interactive prefix idea but unfortunately this doesn't quite emulate the old behavior. `loadfile` will instantly exit mpv since the url is empty but it's optional so it's still not a hard error. How about just dropping the empty url part of the implementation and just letting whatever is selected by the dialog (or not selected) override anything that is passed in the url part of the command?

Reply:
This is fixed now, there was a typo in the check below.

Also `interactive` prefix means that iteration may happen, but not that it will/should happen.